redis/commands.py

```diff
@@ -1173,13 +1173,21 @@ def set(self, name, value,
         if ex is not None:
             pieces.append('EX')
             if isinstance(ex, datetime.timedelta):
-                ex = int(ex.total_seconds())
-            pieces.append(ex)
+                ex = ex.total_seconds()
+            try:
```

Comment:
What about:

if not isinstance(ex, int):
    raise DataError("something something").

Repeat